src/ln/channelmonitor.rs

```diff
@@ -286,12 +286,27 @@ impl ChannelMonitor {
 				}
 			}
 		}
-		// TODO: Prune payment_preimages no longer needed by the revocation (just have to check
-		// that non-revoked remote commitment tx(n) do not need it, and our latest local commitment
-		// tx does not need it.
+
+		let mut pruned_payment_preimages = HashMap::with_capacity(self.payment_preimages.capacity());
+		for (k, v) in self.payment_preimages.drain() {
+			for (htlc, _s1, _s2) in &self.current_local_signed_commitment_tx.as_ref().expect("Channel need at least an initial commitment tx !").htlc_outputs {
+				if k == htlc.payment_hash {
+					pruned_payment_preimages.insert(k, v); 
+				}
+			}
+			for (_tx, htlcs) in &self.remote_claimable_outpoints {
```

Comment:
This seems like its gonna be a pretty large chunk of runtime, but also overkill. Can we effeciently skip remote outpoints that are already revoked somehow, cause we shouldn't need those preimages either, we can use the revocation key.

Reply:
Outpoint is considered as revoked when we get its per_commitment_secret. At pruning, we need to check that its commitment number is superior at get_min_secret(), but how do we get this commitment number without input sequence and lock_time of the given outpoint, it's only place where it's stored.
Will inquiry more tomorrow to find a map between txid:commitment number, correct me if I'm wrong.